### PR TITLE
feat: add react-native-payments

### PR DESCRIPTION
**Pull request by rc23**
## Description

This provides the new react-native-payments package.

add react-native-payments package
add CreditCard WebView logic and test
add README.md

### Dependencies

None.

## Checklist

- [x] The commit message follows our guidelines
- [x] Tests for the respective changes have been added
- [x] The code is commented, particularly in hard-to-understand areas
- [x] The labels and/or milestones were added

## Disclaimer

By sending us your contributions, you are agreeing that your contribution is made subject to the terms of our [Contributor Ownership Statement](https://github.com/Farfetch/.github/blob/master/COS.md)

### CODE REVIEW

**Review by ivomota — APPROVED** [review]

File: packages/react-native-payments/README.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+          staticName: STATIC_NAME,
+          url: PAYMENT_GATEWAY_URL,
+        }}
+        creditCardDispatch={creditCardDispatch}
+        paymentIntentId={checkoutOrder.paymentIntentId}
+        webViewRef={webViewRef}
+      />
+      <S.ButtonView>
+        <Button fullwidth primary onPress={handleCharge}>
+          {t('submit')}
```
Comment (nelsonleite) [inline packages/react-native-payments/README.md:78]:
```suggestion
          submit
```

Nitpick: for the sake of the example I'd remove the translation function 🙂

File: packages/react-native-payments/package.json [hunk 2]
```diff
@@ -0,0 +1,25 @@
+{
+  "name": "react-native-payments",
+  "version": "0.0.0",
+  "description": "Payments solutions for react-native apps.",
+  "main": "src/index.js",
+  "files": [
+    "src"
+  ],
+  "license": "UNLICENSED",
```
Comment (nelsonleite) [inline packages/react-native-payments/package.json:9]:
```suggestion
  "license": "MIT",
```

File: packages/react-native-payments/package.json [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  "publishConfig": {
```
Comment (nelsonleite) [inline packages/react-native-payments/package.json:10]: Just to keep consistency across all packages, please move `publishConfig` to the end and `peerDependencies` after `devDependencies` 😇

File: packages/react-native-payments/src/components/CreditCard.js [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+      }
+
+      const operation = data?.operation;
+
+      if (operation === OPERATION_TYPE.VALIDATION) {
+        creditCardDispatch({ isFormValid: data?.isFormValid });
+      }
+
+      const payload = data?.payload;
+
```
Comment (nelsonleite) [inline packages/react-native-payments/src/components/CreditCard.js:48]: You don't need to leave empty lines on every `const`, you can stick them together 🙂 

Please review this everywhere 🙏

**Review by nelsonleite — CHANGES_REQUESTED** [review]

File: packages/react-native-payments/README.md (hunk 1 above)
Comment (rc23) [reply packages/react-native-payments/README.md]: Should be ok now, please double check 👍

File: packages/react-native-payments/package.json (hunk 2 above)
Comment (rc23) [reply packages/react-native-payments/package.json]: Should be ok now, please double check 👍

File: packages/react-native-payments/package.json (hunk 3 above)
Comment (rc23) [reply packages/react-native-payments/package.json]: Should be ok now, please double check 👍

File: packages/react-native-payments/src/components/CreditCard.js (hunk 4 above)
Comment (rc23) [reply packages/react-native-payments/src/components/CreditCard.js:46]: Should be ok now, please double check 👍

File: packages/react-native-payments/package.json (hunk 3 above)
Comment (nelsonleite) [reply packages/react-native-payments/package.json]: You just missed moving `peerDependencies` below `devDependencies` 😬

File: packages/react-native-payments/src/components/CreditCard.js (hunk 4 above)
Comment (nelsonleite) [reply packages/react-native-payments/src/components/CreditCard.js:46]: Hmm, it still seems that there are unnecessary spaces (lines 46 and 48)

File: packages/react-native-payments/package.json (hunk 3 above)
Comment (rc23) [reply packages/react-native-payments/package.json]: Sorry about that, missed it. Should be ok now 👍

File: packages/react-native-payments/src/components/CreditCard.js (hunk 4 above)
Comment (rc23) [reply packages/react-native-payments/src/components/CreditCard.js]: Sorry about that, missed it. Should be ok now 👍

**Comment by ruifcnunes** [comment]
After checking with @rc23 this PR can be closed. thanks

**Pull request closed without merging**